components/mpas-ocean/cime_config/buildnml

```diff
@@ -74,184 +82,210 @@ def buildnml(case, caseroot, compname):
 
     if ocn_grid == 'oEC60to30':
-        ic_date += '151031'
-        ic_prefix += 'oEC60to30'
-        decomp_date += '151031'
-        decomp_prefix += 'mpas-o.graph.info.'
-    elif ocn_grid == 'oEC60to30_ICG':
-        ic_date += '160827'
-        ic_prefix += 'oEC60to30.restartFrom_eos1b'
         decomp_date += '151031'
         decomp_prefix += 'mpas-o.graph.info.'
+        ic_date += '151031'
+        ic_prefix += 'oEC60to30'
+        if ocn_ic_mode == 'spunup':
+            ic_date += '160827'
+            ic_prefix += 'oEC60to30.restartFrom_eos1b'
+
     elif ocn_grid == 'oEC60to30v3':
+        decomp_date += '161222'
+        decomp_prefix += 'mpas-o.graph.info.'
+        restoring_file += 'sss.monthlyClimatology.PHC2_salx_040803.oEC60to30v3.nc'
+        analysis_mask_file += 'oEC60to30v3_Atlantic_region_and_southern_transect.nc'
         ic_date += '170905'
         ic_prefix += 'oEC60to30v3_60layer'
         #  modify initial condition file if ocean BGC is turned on
         if ocn_bgc in ['eco_only', 'eco_and_dms', 'eco_and_macromolecules', 'eco_and_dms_and_macromolecules']:
             ic_date = '180418'
             ic_prefix = 'mpaso.rst.BCRC_CNPECACNT_1850_anvil02.0066-01-01.no-xtime'
             eco_forcing_file = 'oEC60to30v3.monthlySurfaceForcingBGC.171120.nc'
-        decomp_date += '161222'
-        decomp_prefix += 'mpas-o.graph.info.'
-        restoring_file += 'sss.monthlyClimatology.PHC2_salx_040803.oEC60to30v3.nc'
-        analysis_mask_file += 'oEC60to30v3_Atlantic_region_and_southern_transect.nc'
-    elif ocn_grid == 'oEC60to30v3_ICG':
-        ic_date += '200927'
-        ic_prefix += 'oEC60to30v3.restartFrom_anvilG'
-        decomp_date += '161222'
-        decomp_prefix += 'mpas-o.graph.info.'
-        analysis_mask_file += 'oEC60to30v3_Atlantic_region_and_southern_transect.nc'
+        if ocn_ic_mode == 'spunup':
+            ic_date += '200927'
+            ic_prefix += 'oEC60to30v3.restartFrom_anvilG'
+
     elif ocn_grid == 'oEC60to30wLI':
-        ic_date += '160830'
-        ic_prefix += 'oEC60to30wLI'
         decomp_date += '160830'
         decomp_prefix += 'mpas-o.graph.info.'
+        ic_date += '160830'
+        ic_prefix += 'oEC60to30wLI'
+        if ocn_ic_mode == 'spunup':
+            logger.warning("WARNING: The specified compset is requesting ocean ICs spunup from a G-case")
+            logger.warning("         But no file available for this grid.")
+
     elif ocn_grid == 'oEC60to30v3wLI':
-        ic_date += '171031'
-        ic_prefix += 'oEC60to30v3wLI60lev'
         decomp_date += '170328'
         decomp_prefix += 'mpas-o.graph.info.'
         restoring_file += 'sss.monthlyClimatology.PHC2_salx_040803.oEC60to30v3wLI.nc'
         analysis_mask_file += 'masks_SingleRegionAtlanticWTransportTransects_EC60to30v3wLI_171116.nc'
+        ic_date += '171031'
+        ic_prefix += 'oEC60to30v3wLI60lev'
+        if ocn_ic_mode == 'spunup':
+            ic_date += '171116'
```

Comment:
@jonbob I just tried testing this (`oEC60to30v3wLI`) grid with a spunup compset and the `ic_date` from the if statement is concatenating the previous defined `ic_date`, so the resulting file it's looking for is `oEC60to30v3wLI60levoEC60to30v3wLI60lev.restart_theta_year26.171031171116.nc`. This logic probably affects other grids as well.

Reply:
the `ic_prefix` is getting double-concatenated as well

Reply:
@darincomeau - can you take a look again at this and review? I think we need to get it merged as soon as possible